test/langtools/jdk/javadoc/doclet/testTagOrder/TestTagOrder.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8263043
+ * @summary Add test to verify order of tag output
+ * @library /tools/lib ../../lib
+ * @modules jdk.javadoc/jdk.javadoc.internal.tool
+ * @build toolbox.ToolBox javadoc.tester.*
+ * @run main TestTagOrder
+ */
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import javadoc.tester.JavadocTester;
+import toolbox.ToolBox;
+
+/**
+ * Tests the order of the output of block tags in the generated output.
+ * There is a default order, embodies in the order of declaration of tags in
+ * {@code Tagl;etManager}, but this can be overridden on the command line by
+ * specifying {@code -tag} options in the desired order.
+ */
+public class TestTagOrder extends JavadocTester {
+    public static void main(String... args) throws Exception {
+        TestTagOrder tester = new TestTagOrder();
+        tester.runTests(m -> new Object[] { Path.of(m.getName()) });
+    }
+
+    ToolBox tb = new ToolBox();
+    Path src = Path.of("src");
+    Map<String, String> expect = new LinkedHashMap<>();
+
+    TestTagOrder() throws IOException {
+        tb.writeJavaFiles(src,
+                """
+                    package p;
+                    /** Class C. */
+                    public class C {
+                        /**
+                         * This is method m.
+                         * @param p1 first parameter
+                         * @param p2 second parameter
+                         * @return zero
+                         * @throws IllegalArgumentException well, never
+                         * @since 1.0
+                         * @see <a href="http://example.com">example</a>
```

Comment:
Consider adding `@version` and `@author` standard tags to this test. Although JDK API documentation does not include those, API documentation of other projects might. Adding those tags will improve coverage.

Reply:
Hmmm ...
Actually, my initial version of the code _did_ include `@author`, but not `@version`.  But, it didn't work as expected ... because neither are actually supported on methods, and so they do not appear in the output.  See https://docs.oracle.com/en/java/javase/15/docs/specs/javadoc/doc-comment-spec.html#where-tags-can-be-used

So, if you really think we should test these as well, we will have to extend the test to include comments on a class or interface declaration.

Reply:
I looked at that table before suggesting to add those tags and realized that adding them would complicate the test, hence "consider". You could push the test as is and we will improve it later. It's up to you, Jon.

Reply:
I've improved the test.